### ISSUE TITLE
v1.22.0 re-released?

### ISSUE THREAD

**Issue by gromgit** [hadolint/hadolint#544]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Homebrew maintains a formula for `hadolint` that was just updated on the Linux side (a.k.a. Linuxbrew) shortly after v1.22.0 was released. Regardless of version, users expect to be able to `brew install hadolint` without issues.

### Actual behavior
A user [just reported](https://github.com/Homebrew/linuxbrew-core/issues/22235) that the SHA256 checksum of the release source tarball no longer matches the tested one. As Homebrew's CI always test-builds before commiting changes to formulae, this naturally raised eyebrows.

It looks like v1.22.0 was re-released at some point. Is that true? If so, is the currently-released commit (4160463da15d08201a15e987af0eb481f395948d) the definitive one for this version, or will there be further releases under the same version tag?

### Steps to reproduce the behavior

```bash
# on a Linuxbrew platform
$ brew install hadolint
==> Downloading https://linuxbrew.bintray.com/bottles/ghc-8.10.3.x86_64_li
Already downloaded: /home/bulbitain/.cache/Homebrew/downloads/8dc81480cd7a40fa8f42ece9b34adeb6ce9c71bea6a193265d1c059bac9f9d1e--ghc-8.10.3.x86_64_linux.bottle.tar.gz
==> Downloading https://linuxbrew.bintray.com/bottles/haskell-stack-2.5.1.
Already downloaded: /home/bulbitain/.cache/Homebrew/downloads/7cee952bfec02bb575f830f7e7e33fb1e5c7e91155068e6788e79990a7cb30ee--haskell-stack-2.5.1.x86_64_linux.bottle.tar.gz
==> Downloading https://github.com/hadolint/hadolint/archive/v1.22.0.tar.g
==> Downloading from https://codeload.github.com/hadolint/hadolint/tar.gz/
Error: SHA256 mismatch
Expected: ab208e7f4d321a43fb9a3d94a26c59afcd7dd5e2e5862a90d38af99504a45403
  Actual: d65d3aafa517a0e234c40b1f3085ef82319ec0f43698dd905080c6f3bb4b83bc
    File: /home/bulbitain/.cache/Homebrew/downloads/1f67bd39df3ea3faf17c4c2b3d5134863e5632c50ba38a9bd8316ae7a6a0d0f6--hadolint-1.22.0.tar.gz
To retry an incomplete download, remove the file above.
```

**Comment by lorenzo** [comment]
Yes, something about a dependency in github actions changed so the build broke in the mid of the process and I had to re-tag after changing the release script. 

I'm sorry for the inconvenience.

**Comment by lorenzo** [comment]
I tagged a new version to get out of this mess. Again, sorry for the confusion.

**Comment by yermulnik** [comment]
@lorenzo Apologies for dumb question, I'm not very familiar with brew: when the change is going to end up being available? I'm still getting sha256 mismatch error:
```
> brew upgrade
==> Upgrading 1 outdated package:
hadolint 1.21.0 -> 1.22.0
==> Upgrading hadolint 1.21.0 -> 1.22.0
==> Downloading https://github.com/hadolint/hadolint/archive/v1.22.0.tar.gz
==> Downloading from https://codeload.github.com/hadolint/hadolint/tar.gz/v1.22.0
######################################################################## 100.0%
Error: SHA256 mismatch
Expected: ab208e7f4d321a43fb9a3d94a26c59afcd7dd5e2e5862a90d38af99504a45403
  Actual: d65d3aafa517a0e234c40b1f3085ef82319ec0f43698dd905080c6f3bb4b83bc
    File: /home/yz/.cache/Homebrew/downloads/1f67bd39df3ea3faf17c4c2b3d5134863e5632c50ba38a9bd8316ae7a6a0d0f6--hadolint-1.22.0.tar.gz
To retry an incomplete download, remove the file above.
```

**Comment by lorenzo** [comment]
I'm not familiar with brew either, so I think we need to wait for brew to publish `v1.22.1`

**Comment by yermulnik** [comment]
Getting build failure with `v1.22.1` – is this brew-related or something local to my OS? (previously `hadolint v1.21.0` installed with brew w/o any issues):
```
> brew upgrade
Updating Homebrew...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
Updated 2 formulae.

==> Upgrading 1 outdated package:
hadolint 1.21.0 -> 1.22.1
==> Upgrading hadolint 1.21.0 -> 1.22.1
==> Downloading https://github.com/hadolint/hadolint/archive/v1.22.1.tar.gz
==> Downloading from https://codeload.github.com/hadolint/hadolint/tar.gz/v1.22.1
######################################################################## 100.0%
==> stack -j4 build
Last 15 lines from /home/yz/.cache/Homebrew/Logs/hadolint/01.stack:
/tmp/hadolint-20210208-16352-xijmhp/hadolint-1.22.1/.brew_home/.stack/programs/x86_64-linux/ghc-8.8.4/lib/ghc-8.8.4/unix-2.7.2.2/libHSunix-2.7.2.2.a(User.o):ghc_2.c:function ghczuwrapperZC0ZCunixzm2zi7zi2zi2ZCSystemziPosixziUserZCendpwent: warning: Using 'endpwent' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/hadolint-20210208-16352-xijmhp/hadolint-1.22.1/.brew_home/.stack/programs/x86_64-linux/ghc-8.8.4/lib/ghc-8.8.4/unix-2.7.2.2/libHSunix-2.7.2.2.a(User.o):ghc_2.c:function ghczuwrapperZC1ZCunixzm2zi7zi2zi2ZCSystemziPosixziUserZCsetpwent: warning: Using 'setpwent' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/hadolint-20210208-16352-xijmhp/hadolint-1.22.1/.brew_home/.stack/programs/x86_64-linux/ghc-8.8.4/lib/ghc-8.8.4/unix-2.7.2.2/libHSunix-2.7.2.2.a(User.o):ghc_2.c:function ghczuwrapperZC2ZCunixzm2zi7zi2zi2ZCSystemziPosixziUserZCgetpwent: warning: Using 'getpwent' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/hadolint-20210208-16352-xijmhp/hadolint-1.22.1/.brew_home/.stack/programs/x86_64-linux/ghc-8.8.4/lib/ghc-8.8.4/unix-2.7.2.2/libHSunix-2.7.2.2.a(User.o):ghc_2.c:function ghczuwrapperZC3ZCunixzm2zi7zi2zi2ZCSystemziPosixziUserZCgetpwnamzur: warning: Using 'getpwnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/hadolint-20210208-16352-xijmhp/hadolint-1.22.1/.brew_home/.stack/programs/x86_64-linux/ghc-8.8.4/lib/ghc-8.8.4/unix-2.7.2.2/libHSunix-2.7.2.2.a(User.o):ghc_2.c:function ghczuwrapperZC4ZCunixzm2zi7zi2zi2ZCSystemziPosixziUserZCgetpwuidzur: warning: Using 'getpwuid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/hadolint-20210208-16352-xijmhp/hadolint-1.22.1/.brew_home/.stack/programs/x86_64-linux/ghc-8.8.4/lib/ghc-8.8.4/unix-2.7.2.2/libHSunix-2.7.2.2.a(User.o):ghc_2.c:function ghczuwrapperZC5ZCunixzm2zi7zi2zi2ZCSystemziPosixziUserZCgetgrnamzur: warning: Using 'getgrnam_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/tmp/hadolint-20210208-16352-xijmhp/hadolint-1.22.1/.brew_home/.stack/programs/x86_64-linux/ghc-8.8.4/lib/ghc-8.8.4/unix-2.7.2.2/libHSunix-2.7.2.2.a(User.o):ghc_2.c:function ghczuwrapperZC6ZCunixzm2zi7zi2zi2ZCSystemziPosixziUserZCgetgrgidzur: warning: Using 'getgrgid_r' in statically linked applications requires at runtime the shared libraries from the glibc version used for linking
/usr/lib/x86_64-linux-gnu/libpthread.a(pthread_cond_broadcast.o)(.note.stapsdt+0x14): error: relocation refers to local symbol "" [1], which is defined in a discarded section
collect2: error: ld returned 1 exit status
`gcc-5' failed in phase `Linker'. (Exit code: 1)
Completed 44 action(s).

--  While building package hadolint-1.22.1 (scroll up to its section to see the error) using:
      /tmp/hadolint-20210208-16352-xijmhp/hadolint-1.22.1/.brew_home/.stack/setup-exe-cache/x86_64-linux/Cabal-simple_mPHDZzAJ_3.0.1.0_ghc-8.8.4 --builddir=.stack-work/dist/x86_64-linux/Cabal-3.0.1.0 build lib:hadolint exe:hadolint --ghc-options ""
    Process exited with code: ExitFailure 1

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
hadolint 1.22.0 fails to install, SHA256 mismatch https://github.com/Homebrew/linuxbrew-core/issues/22235
```

Thanks in advance for any clue.

**Comment by lorenzo** [comment]
It looks like a brew problem. A thing that changed from the previous release: static binary builds are enabled by default.

They should not be affecting OS X, though, as steric canaries are not possible there, but maybe it is contributing to the problem somehow.

**Comment by lorenzo** [comment]
Oh, that’s it! You are using homebrew in Linux, and the errors are about the static compilation. I recommend you pass the cabal flag “-f-static” when building.

**Comment by yermulnik** [comment]
@lorenzo Thanks for the pointer. I'm going to try and build with the flag you suggested.

> A thing that changed from the previous release: static binary builds are enabled by default

Is this `hadolint` or `brew` which has changed?

**Comment by yermulnik** [comment]
Oh, it seems like `brew` already has precompiled binaries:
```
> brew upgrade hadolint
==> Upgrading 1 outdated package:
hadolint 1.21.0 -> 1.22.1
==> Upgrading hadolint 1.21.0 -> 1.22.1
==> Downloading https://linuxbrew.bintray.com/bottles/hadolint-1.22.1.x86_64_linux.bottle.tar.gz
==> Downloading from https://akamai.bintray.com/d3/d3060f651b65a8bd3213b1378ba753cd76fdea01b5e5937ba57d0fb43b97f859?__gd
######################################################################## 100.0%
==> Pouring hadolint-1.22.1.x86_64_linux.bottle.tar.gz
🍺  /home/linuxbrew/.linuxbrew/Cellar/hadolint/1.22.1: 5 files, 21.4MB
Removing: /home/linuxbrew/.linuxbrew/Cellar/hadolint/1.21.0... (5 files, 19.7MB)
Removing: /home/yz/.cache/Homebrew/hadolint--1.21.0.tar.gz... (112.6KB)
```

Looks like it attempted to build from source for me yesterday because of this 🤷 
Anyways thanks for helping!
